apps/web/components/Navbar.tsx

```diff
@@ -33,20 +34,44 @@ export const ConnectWalletButton = () => {
   };
 
   return (
-    <>
+    <div
+      onMouseEnter={() => setShowDropdown(true)} // Show dropdown on hover
+      onMouseLeave={() => setShowDropdown(false)} // Hide dropdown when not hovering
+      className="relative" // Ensure dropdown is positioned relative to this container
+    >
       {connected ? (
-        <div>
-          {formatAddress(account)}{" "}
-          {/* Using the formatAddress utility function */}
-          <button onClick={disconnect}>DISCONNECT</button>{" "}
-          {/* Disconnect button */}
-        </div>
+        <>
+          <Button
+            variant="primary"
+            onClick={() => setShowDropdown(!showDropdown)}
+          >
+            {formatAddress(account)}
+          </Button>
+          {showDropdown && ( // Conditionally render dropdown based on state
```

Comment:
As I remember this was pointed out in the search input PR as well. We could easily use a Popover instead of using state for this. which is better that relying on `useState`
https://www.radix-ui.com/primitives/docs/components/popover

Reply:
@chin-flags  I remember using this back then.
https://ui.shadcn.com/docs/components/popover
I will do that.
Thank you so much for pointing it out, I need to understand more where to use this popover rather than state.